httplint/field/cors.py

```diff
@@ -0,0 +1,99 @@
+from typing import cast, TYPE_CHECKING
+
+from httplint.note import Note, categories, levels
+from httplint.types import AddNoteMethodType
+
+if TYPE_CHECKING:
+    from httplint.message import HttpMessageLinter, HttpRequestLinter
+
+
+def check_preflight_request_header(
+    message: "HttpMessageLinter", canonical_name: str, add_note: AddNoteMethodType
+) -> None:
+    """
+    Check if the header is only present in a message that is a CORS preflight request.
+    """
+    # pylint: disable=import-outside-toplevel
+    from httplint.message import HttpRequestLinter
+
+    if isinstance(message, HttpRequestLinter):
+        if message.method != "OPTIONS":
+            add_note(
+                CORS_PREFLIGHT_REQ_METHOD_WRONG,
+                field_name=canonical_name,
+                method=message.method,
+            )
+        if "origin" not in message.headers.parsed:
+            add_note(CORS_PREFLIGHT_REQ_NO_ORIGIN, field_name=canonical_name)
+        if (
+            canonical_name == "Access-Control-Request-Headers"
+            and "access-control-request-method" not in message.headers.parsed
+        ):
```

Comment:
I don't really understand this logic. I don't think either of these headers is required to be present, but why are they compared in different casing here? And why are they in the same check? Do they have individual checks elsewhere?